core/common/src/ContourDisplay.ts

```diff
@@ -0,0 +1,309 @@
+      return diff;
+
+    return diff;
+  }
+
+  private constructor(json?: ContourProps) {
+    if (undefined === json) {
+      this.majorStyle = ContourStyle.fromJSON({ pixelWidth: 2 });
+      this.minorStyle = ContourStyle.fromJSON();
+      this.minorInterval = 1;
+      this.majorIntervalCount = 5;
+      this.showGeometry = true;
+    } else {
+      this.majorStyle = json.majorStyle ? ContourStyle.fromJSON(json.majorStyle) : ContourStyle.fromJSON({ pixelWidth: 2 });
+      this.minorStyle = json.minorStyle ? ContourStyle.fromJSON(json.minorStyle) : ContourStyle.fromJSON();
+      this.minorInterval = json.minorInterval ?? 1;
+      this.majorIntervalCount = json.majorIntervalCount ?? 5;
+      this.showGeometry = json.showGeometry ?? true;
+    }
+  }
+
+  public static fromJSON(json?: ContourProps) {
+    return json ? new Contour(json) : new Contour({});
+  }
+
+  public toJSON(): ContourProps {
+    const props: ContourProps = {};
+
+    if (!this.majorStyle.equals(ContourStyle.fromJSON({ pixelWidth: 2 })))
+      props.majorStyle = this.majorStyle.toJSON();
+
+    if (!this.minorStyle.equals(ContourStyle.fromJSON()))
+      props.minorStyle = this.minorStyle.toJSON();
+
+    if (1 !== this.minorInterval)
+      props.minorInterval = this.minorInterval;
+
+    if (5 !== this.majorIntervalCount)
+      props.majorIntervalCount = this.majorIntervalCount;
+
+    if (true !== this.showGeometry)
+      props.showGeometry = this.showGeometry;
+
+    return props;
+  }
+}
+
+/** JSON representation of the description of how contours should appear for a particular set of subcategories. */
+export interface ContourGroupProps {
+  /** See [[ContourGroup.contourDef]]. */
+  contourDef?: ContourProps;
+  /** See [[ContourGroup.subCategories]]. */
+  subCategories?: CompressedId64Set;
+}
+
+/** Contains a description of how contours should appear for a particular set of subcategories. A contour group is an organizational concept which associates a contour appearance with a list of subcategories within an iModel.
+   * @see [[Contour]]
+   */
+export class ContourGroup {
+  private _subCategories: CompressedId64Set;
+
+  /** A [[Contour]] object describing how the contours for this contour group should appear. Defaults to an instantation of [[Contour]] using all of its own default properties. */
+  public readonly contourDef: Contour;
+
+  /** List of subcategory IDs to which this contour group's styling will be applied. */
+  public get subCategories(): OrderedId64Iterable {
+    return CompressedId64Set.iterable(this._subCategories);
+  }
+
+  public equals(other: ContourGroup | undefined): boolean {
+    if (this === undefined && other === undefined)
+      return true;
+    if (this === undefined || other === undefined)
+      return false;
+    if (!this.contourDef.equals(other.contourDef))
+      return false;
+    let subCount = 0;
+    let otherSubCount = 0;
+    for (const subCategory of this.subCategories) {
```

Comment:
@pmconne This equality comparison nested loop is not ideal compared to the way it used to be before the latest API changes I pushed. Is it safe for us to just compare the underlying `CompressedId64Set` as a string against the other one?

Reply:
Yes, that's what I meant by making comparison trivial. Two CompressedId64Sets contain the same Ids if and only if they are identical strings.

Reply:
Ok, thanks for verifying. The enforced ordering makes a lot of sense in that context. I'll change that comparison logic and resolve this comment when I push it.

Reply:
I fixed the subcategory comparison.